src/core/execute.c

```diff
@@ -514,6 +529,13 @@ int exec_spawn(
                 }
         }
 
+        /* We cannot spawn the main service process into the subcgroup as it might need to unshare the cgroup
+         * namespace first if one is configured to make sure the root of the cgroup namespace is the service
+         * cgroup and not the subcgroup. However, when running control commands on a live service, the
+         * commands have to be spawned inside a subcgroup, otherwise we violate the no inner processes rule
+         * of cgroupv2. */
+        const char *cgtarget = exec_params_needs_control_subcgroup(params) ? subcgroup_path : params->cgroup_path;
```

Comment:
i.e. the cg_create shall only be called if exec_params_needs_control_subcgroup.

And exec_invoke needs to be switched to `cg_create_and_attach`

Reply:
Can't use cg_create_and_attach(), it's easier to always make sure the subcgroup is created early rather than creating it on demand.